### DISCUSSION TITLE
Use anyhow

### DISCUSSION THREAD

**Discussion by dlen** [apalis-dev/apalis#258] [General]
Hello!

I'm trying to update my code to version 0.5 but I'm having issues using anyhow.

I managed to reproduce it using the code in the examples. Just modifying the send_email signature:

```
pub async fn send_email(job: Email) -> anyhow::Result<()> {
    log::info!("Attempting to send email to {}", job.to);
    Ok(())
}
```

Outputs the following when building the actix-web example:
```
error[E0277]: the trait bound `ServiceFn<fn(Email) -> impl futures::Future<Output = Result<(), anyhow::Error>> {send_email}, _>: tower_service::Service<apalis::prelude::Request<_>>` is not satisfied
  --> examples/actix-web/src/main.rs:48:18
   |
48 |                 .build_fn(send_email)
   |                  ^^^^^^^^ the trait `tower_service::Service<apalis::prelude::Request<_>>` is not implemented for `ServiceFn<fn(Email) -> impl futures::Future<Output = Result<(), anyhow::Error>> {s
end_email}, _>`
```

I'm not sure if this is still possible to do or I'm missing something. In any case thanks a lot for the efforts in building this library!!

**Comment by geofmureithi** [comment]
As of 0.5, the error returned by the job must impl `std::error::Error`.
Unfortunately `anyhow::Error` does not `impl std::error::Error`. You should consider using `Box<Error>` or manually impl error, or a crate like thiserror. 
See https://github.com/dtolnay/anyhow/issues/25

Rust doesn't have specialization yet, so this is not possible currently.

**Reply by dlen** [reply]
Thanks for the clarification. I have the impression though that this was possible in 0.4.x.
